infra/base-images/base-runner/bad_build_check

```diff
@@ -62,78 +62,69 @@ function check_startup_crash {
     CHECK_PASSED=1
   fi
 
-  if (( $CHECK_PASSED == 0 )); then
-    echo "BAD BUILD: the fuzzer seems to have either startup crash or exit."
-    exit 1
+  if [ "$CHECK_PASSED" -eq "0" ]; then
+    echo "BAD BUILD: the fuzzer seems to have either startup crash or exit." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 }
 
 # Mixed sanitizers check for ASan build.
 function check_asan_build {
   local FUZZER=$1
-  local ASAN_CALLS=$2
-  local MSAN_CALLS=$3
-  local UBSAN_CALLS=$4
-
-  # Perform all the checks for more verbose error message.
-  local CHECK_FAILED=0
+  local LOG_PATH_FOR_BROKEN_TARGET=$2
+  local ASAN_CALLS=$3
+  local MSAN_CALLS=$4
+  local UBSAN_CALLS=$5
 
+  # Perform all the checks for more detailed error message.
   if (( $ASAN_CALLS < 1000 )); then
-    echo "BAD BUILD: $FUZZER does not seem to be compiled with ASan."
-    CHECK_FAILED=1
+    echo "BAD BUILD: $FUZZER does not seem to be compiled with ASan." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 
   if (( $MSAN_CALLS > 0 )); then
-    echo "BAD BUILD: ASan build of $FUZZER seems to be compiled with MSan."
-    CHECK_FAILED=1
+    echo "BAD BUILD: ASan build of $FUZZER seems to be compiled with MSan." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 
   if (( $UBSAN_CALLS > 250 )); then
-    echo "BAD BUILD: ASan build of $FUZZER seems to be compiled with UBSan."
-    CHECK_FAILED=1
-  fi
-
-  if (( $CHECK_FAILED > 0 )); then
-    exit 1
+    echo "BAD BUILD: ASan build of $FUZZER seems to be compiled with UBSan." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 }
 
 # Mixed sanitizers check for MSan build.
 function check_msan_build {
   local FUZZER=$1
-  local ASAN_CALLS=$2
-  local MSAN_CALLS=$3
-  local UBSAN_CALLS=$4
-
-  # Perform all the checks for more verbose error message.
-  local CHECK_FAILED=0
+  local LOG_PATH_FOR_BROKEN_TARGET=$2
+  local ASAN_CALLS=$3
+  local MSAN_CALLS=$4
+  local UBSAN_CALLS=$5
 
+  # Perform all the checks for more detailed error message.
   if (( $ASAN_CALLS > 0 )); then
-    echo "BAD BUILD: MSan build of $FUZZER seems to be compiled with ASan."
-    CHECK_FAILED=1
+    echo "BAD BUILD: MSan build of $FUZZER seems to be compiled with ASan." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 
   if (( $MSAN_CALLS < 1000 )); then
-    echo "BAD BUILD: $FUZZER does not seem to be compiled with UBSan."
-    CHECK_FAILED=1
+    echo "BAD BUILD: $FUZZER does not seem to be compiled with UBSan." >> $LOG_PATH_FOR_BROKEN_TARGET
+    BAD_BUILD_CHECK_FAILED=1
   fi
 
   if (( $UBSAN_CALLS > 250 )); then
```

Comment:
Shouldn't be this be > 0

Reply:
Unfortunately no :/ As I remember, there always are some `ubsan` calls even if you use only ASan / MSan. I'll double check before landing.